### PR TITLE
Added Separate Field In Form To Add Optional Birth Time (Issue #445)

### PR DESCRIPTION
**Pull request by krooldonutz**
Regarding Ticket [#445](https://github.com/josdem/vetlog-spring-boot/issues/445)

This issue is also happening to me in Chrome and Safari.

My proposed solution is to separate the forms from birthDateTime to birthDate and birthTIme, which would be combined when the date is being validated

### CODE REVIEW

**Comment by josdem** [comment]
> Regarding Ticket [#445](https://github.com/josdem/vetlog-spring-boot/issues/445)
> 
> This issue is also happening to me in Chrome and Safari.
> 
> My proposed solution is to separate the forms from birthDateTime to birthDate and birthTIme, which would be combined when the date is being validated

Hi @krooldonutz thank you for providing a solution; however, we do not really need a time for a birthdate, so we will be okay if we just remove the time of birth.

**Comment by krooldonutz** [comment]
Hello @josdem, do you want to remove the time of birth from the form or from the entirety of the codebase as well (database, and etc)? if it's only from the form, I would just default it to 6am or 00:00

**Comment by josdem** [comment]
> Hello @josdem, do you want to remove the time of birth from the form or from the entirety of the codebase as well (database, and etc)? if it's only from the form, I would just default it to 6am or 00:00

Hi @krooldonutz let's just default time value to 6:00 am, thank you!

**Comment by krooldonutz** [comment]
Okay, I'll do it after I get off work 😊

**Comment by krooldonutz** [comment]
Hello @josdem , please take a look at the new commit 😆

File: src/main/java/com/josdem/vetlog/command/PetCommand.java [hunk 1]
```diff
@@ -23,6 +23,8 @@
 import jakarta.validation.constraints.Size;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
+
 import lombok.Getter;
```
Comment (josdem) [inline src/main/java/com/josdem/vetlog/command/PetCommand.java:28]: Please delete this unused import

File: src/main/resources/i18n/messages.properties [hunk 2]
```diff
@@ -70,6 +70,7 @@ pet.view.edit.title=Edit a Pet
 pet.register=Register a pet
 pet.name=Pet's Name
 pet.birthDate=Date of Birth
+pet.birthTime=Time of Birth (optional)
 pet.dewormed=Dewormed
```
Comment (josdem) [inline src/main/resources/i18n/messages.properties:74]: Also this message value seems not used in the code.

File: src/main/resources/templates/pet/create.html [hunk 3]
```diff
@@ -29,7 +35,7 @@
                             <input type="file" name="image"/>
                             <br/>
                             <label for="birthDate"><h4 th:text="#{pet.birthDate}"/></label>
-                            <input th:data-testid="birthDate" type="datetime-local" id="birthDate" name="birthDate">
+                            <input th:data-testid="birthDate" type="date" id="birthDate" name="birthDate" min="${#dates.format(#dates.createNow(), 'yyyy-MM-dd')}">
```
Comment (josdem) [inline src/main/resources/templates/pet/create.html:38]: Nice :+1:

**Review by josdem — CHANGES_REQUESTED** [review]
Nice PR @krooldonutz, just some small changes before merge.

File: src/main/java/com/josdem/vetlog/command/PetCommand.java (hunk 1 above)
Comment (krooldonutz) [reply src/main/java/com/josdem/vetlog/command/PetCommand.java]: done

File: src/main/resources/i18n/messages.properties (hunk 2 above)
Comment (krooldonutz) [reply src/main/resources/i18n/messages.properties]: and done

**Comment by krooldonutz** [comment]
@josdem finished the requested changes

**Review by kaminuma — APPROVED** [review]

**Comment by josdem** [comment]
> @josdem finished the requested changes

Thanks for doing suggested changes :+1:

**Comment by josdem** [comment]
@all-contributors please add @krooldonutz for code

**Review by josdem — APPROVED** [review]

**Review by kaminuma — APPROVED** [review]
Hi @josdem 
It is my understanding that by changing to date type in the previous merge, a type mismatch occurred in the datetime type part of the test case, and the test did not pass. Although I am currently unable to confirm the operation, I believe that modifying the test code will not cause the above problem.

**Pull request merged**